homeassistant/components/tuya/light.py

```diff
@@ -104,6 +104,7 @@ class TuyaLightEntityDescription(LightEntityDescription):
             brightness=(DPCode.BRIGHT_VALUE_V2, DPCode.BRIGHT_VALUE),
             color_temp=(DPCode.TEMP_VALUE_V2, DPCode.TEMP_VALUE),
             color_data=(DPCode.COLOUR_DATA_V2, DPCode.COLOUR_DATA),
+            separate_brightness_message=True,
```

Comment:
This is assuming all lights need a separate message?

> Bugfix for Tuya dimmers that now require two separate messages to turn on, and set brightness.

Does this apply to all `dj` category lights? How would this apply to all `dj` lights and not all `dd` lights?

Is there a reference I can look at that describes this change in behavior?

Reply:
Hi @frenck.

By default, all categories have the `separate_brightness_message` flag set to false - so existing "single message" behavior is retained by default.

<img width="561" alt="image" src="https://user-images.githubusercontent.com/2313681/183744329-f2a1859d-c388-4d29-8171-552dfca057dd.png">

We set the flag to `true` *only* for the `dj` category of lights:

<img width="758" alt="image" src="https://user-images.githubusercontent.com/2313681/183744516-8a7e94d1-5207-4199-a45b-a9dd18564bef.png">

I've not been able to find any documentation for this change of behavior from Tuya/Gosund/etc. I only have bug reports that dimming does not work (see linked issues) - along with my observation that dimming does not work if brightness & power values are sent in the same message, but does if separately.

Reply:
The Tuya instruction set is here: https://developer.tuya.com/en/docs/iot/dj?id=K9i5ql3v98hn3 however I cannot find any specific information on whether the commands for switch & brightness are expected to work when sent together... real-world testing says 'no, they need to be separate'

Reply:
> real-world testing says 'no, they need to be separate'

I have a serious amount of Tuya devices for testing: none of them show this behavior your describing. So, the code suggestion to say all `dj` lights need separate commands, is incorrect. Also, I don't see why `dd` would be different compared to `dj`.

The current implementation seems to be following the implementation as documented. Did you contact Tuya about the issue? To me, this sounds like a bug in their processing. What did they say?

Reply:
The specific devices I've observed the behavior on are these guys: 

https://us.gosund.com/products/gosund-smart-dimmer-switch-sw2-4-pack

It's the only ones I have to test with (and this change resolves my issue, however I acknowledge the suggested change may not be necessary for other devices.

Without broader range of devices to test against, seems like I'm not in a position to offer a code change :) - will cancel PR

Reply:
I dunno, I think you are still up to something here.

Preferably send out a message to Tuya support about it (in your Tuya IoT Cloud account).

My main concern is the weird separation of the `separate_brightness_message` in this case. We could also consider removing that flag and just always send it separately for all lights (and not just `dj`).

This PR has one potential downside: In some cases, it might cause your light to turn on without it being set to the specified brightness (but the previous one instead) and will start dimming/brightening after that.

Reply:
I think generally speaking, if there are concerns about adverse effects of code changes, with limited ability to test - the best way to proceed is to make the change to behavior as specific as possible.

The alternative would be to somehow identify the device more specifically than just by 'category'  (e.g. read device info and identify combination of category/manufacturer/model/something) and only match the combination of values that exhibits problems.

I'll send a message to Tuya support first, though - and report back.